QuantConnect.ExanteBrokerage/ExanteBrokerageFactory.cs

```diff
@@ -0,0 +1,131 @@
+            { "exante-application-id", Config.Get("exante-application-id") },
+            { "exante-shared-key", Config.Get("exante-shared-key") },
+            { "exante-account-id", Config.Get("exante-account-id") },
+            { "exante-platform-type", Config.Get("exante-platform-type") },
+        };
+
+        public ExanteBrokerageFactory() : base(typeof(ExanteBrokerage))
+        {
+        }
+
+        /// <summary>
+        /// Gets a brokerage model that can be used to model this brokerage's unique behaviors
+        /// </summary>
+        /// <param name="orderProvider">The order provider</param>
+        public override IBrokerageModel GetBrokerageModel(IOrderProvider orderProvider)
+        {
+            return new ExanteBrokerageModel();
+        }
+
+        public static ExanteClientOptions CreateExanteClientOptions(
+            string clientId,
+            string applicationId,
+            string sharedKey,
+            string platformTypeStr
+        )
+        {
+            var platformTypeParsed = Enum.TryParse(platformTypeStr, true, out ExantePlatformType platformType);
+            if (!platformTypeParsed)
+            {
+                throw new Exception($"ExantePlatformType parse error: {platformTypeStr}");
+            }
+
+            var exanteClientOptions =
+                new ExanteClientOptions(
+                    new ExanteApiCredentials(
+                        clientId,
+                        applicationId,
+                        sharedKey
+                    ),
+                    platformType
+                );
+            return exanteClientOptions;
+        }
+
+        /// <summary>
+        /// Creates a new IBrokerage instance
+        /// </summary>
+        /// <param name="job">The job packet to create the brokerage for</param>
+        /// <param name="algorithm">The algorithm instance</param>
+        /// <returns>A new brokerage instance</returns>
+        public override IBrokerage CreateBrokerage(LiveNodePacket job, IAlgorithm algorithm)
+        {
+            var errors = new List<string>();
+
+            // read values from the brokerage data
+            var clientId = Read<string>(job.BrokerageData, "exante-client-id", errors);
+            var applicationId = Read<string>(job.BrokerageData, "exante-application-id", errors);
+            var sharedKey = Read<string>(job.BrokerageData, "exante-shared-key", errors);
+            var accountId = Read<string>(job.BrokerageData, "exante-account-id", errors);
+            var platformTypeStr = Read<string>(job.BrokerageData, "exante-platform-type", errors);
+
+            if (!errors.empty())
+            {
+                // if we had errors then we can't create the instance
+                throw new Exception(string.Join(System.Environment.NewLine, errors));
+            }
+
+            var clientOptions = CreateExanteClientOptions(clientId, applicationId, sharedKey, platformTypeStr);
```

Comment:
I guess it would be more generic if it were ExanteSettings or something similar and we just passed that instance around, will make the API cleaner I think?

Reply:
fixed by 60e49eae6b34581c57e29eaf6feb96800344ea22